invenio_app_ils/ui/src/invenio_app_ils/common/components/ESSelector/serializer.js

```diff
@@ -0,0 +1,95 @@
+const formatPid = pid => `PID: ${pid}`;
+
+export const serializeDocument = metadata => ({
+  id: metadata.document_pid,
+  key: metadata.document_pid,
+  title: metadata.title,
+  description: metadata.authors
+    ? `Authors: ${metadata.authors.join(', ')}`
+    : 'Document',
+  extra: formatPid(metadata.document_pid),
+  metadata: metadata,
+});
+
+export const serializeInternalLocation = metadata => ({
+  id: metadata.internal_location_pid,
+  key: metadata.internal_location_pid,
+  title: metadata.name,
+  description: metadata.location ? `Location: ${metadata.location.name}` : null,
+  extra: formatPid(metadata.internal_location_pid),
+  metadata: metadata,
+});
+
+export const serializeItem = metadata => ({
+  id: metadata.item_pid,
+  key: metadata.item_pid,
+  title: metadata.medium,
+  description: metadata.shelf ? `Shelf: ${metadata.shelf}` : null,
+  extra: formatPid(metadata.item_pid),
+  metadata: metadata,
+});
+
+export const serializeKeyword = metadata => ({
+  id: metadata.keyword_pid,
+  key: metadata.keyword_pid,
+  title: metadata.name,
+  description: metadata.provenance
+    ? `Provenance: ${metadata.provenance}`
+    : null,
+  extra: formatPid(metadata.keyword_pid),
+  metadata: metadata,
+});
+
+export const serializeLoan = metadata => ({
+  id: metadata.loan_pid,
+  key: metadata.loan_pid,
+  title: metadata.state,
+  description: metadata.document_pid
+    ? `Document PID: ${metadata.document_pid}`
+    : null,
+  extra: formatPid(metadata.loan_pid),
+  metadata: metadata,
+});
+
+export const serializeLocation = metadata => ({
+  id: metadata.location_pid,
+  key: metadata.location_pid,
+  title: metadata.name,
+  description: metadata.address ? `Address: ${metadata.address}` : null,
+  extra: formatPid(metadata.location_pid),
+  metadata: metadata,
+});
+
+export const serializePatron = metadata => ({
+  id: metadata.id,
+  key: metadata.id,
+  title: metadata.email,
+  description: metadata.name ? `Name: ${metadata.name}` : null,
+  extra: `ID: ${metadata.id}`,
+  metadata: metadata,
+});
+
+export const serializeHit = hit => {
+  const { metadata } = hit;
+
+  let result = {};
+  if (metadata.loan_pid) {
```

Comment:
I think this way of identifying which type of a record to serialise it is a bit risky, especially that you identify the patron by the email. Maybe we can use the schemas name? or some param passed ?

Reply:
I agree it's risky and will not be very stable if we make changes in the future. The schema name would work great except for patrons which don't have a schema but I suppose if we add a param to patrons we could easily identify those as well.

If we had access to the index name for each hit it would be great but I'm not sure ES supports returning those.